src/lib.rs

```diff
@@ -95,56 +95,40 @@ impl SocketType {
     }
 }
 
-/// Socket Events
-#[allow(non_camel_case_types)]
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub enum SocketEvent {
-    // TODO: This should become a proper enum, including the data.
-    CONNECTED = zmq_sys::ZMQ_EVENT_CONNECTED as isize,
-    CONNECT_DELAYED = zmq_sys::ZMQ_EVENT_CONNECT_DELAYED as isize,
-    CONNECT_RETRIED = zmq_sys::ZMQ_EVENT_CONNECT_RETRIED as isize,
-    LISTENING = zmq_sys::ZMQ_EVENT_LISTENING as isize,
-    BIND_FAILED = zmq_sys::ZMQ_EVENT_BIND_FAILED as isize,
-    ACCEPTED = zmq_sys::ZMQ_EVENT_ACCEPTED as isize,
-    ACCEPT_FAILED = zmq_sys::ZMQ_EVENT_ACCEPT_FAILED as isize,
-    CLOSED = zmq_sys::ZMQ_EVENT_CLOSED as isize,
-    CLOSE_FAILED = zmq_sys::ZMQ_EVENT_CLOSE_FAILED as isize,
-    DISCONNECTED = zmq_sys::ZMQ_EVENT_DISCONNECTED as isize,
-    MONITOR_STOPPED = zmq_sys::ZMQ_EVENT_MONITOR_STOPPED as isize,
-    HANDSHAKE_FAILED_NO_DETAIL = zmq_sys::ZMQ_EVENT_HANDSHAKE_FAILED_NO_DETAIL as isize,
-    HANDSHAKE_SUCCEEDED = zmq_sys::ZMQ_EVENT_HANDSHAKE_SUCCEEDED as isize,
-    HANDSHAKE_FAILED_PROTOCOL = zmq_sys::ZMQ_EVENT_HANDSHAKE_FAILED_PROTOCOL as isize,
-    HANDSHAKE_FAILED_AUTH = zmq_sys::ZMQ_EVENT_HANDSHAKE_FAILED_AUTH as isize,
-    ALL = zmq_sys::ZMQ_EVENT_ALL as isize,
-}
-
-impl SocketEvent {
-    pub fn to_raw(self) -> u16 {
-        self as u16
-    }
-
-    // TODO: this should not need to be public
-    pub fn from_raw(raw: u16) -> SocketEvent {
-        use SocketEvent::*;
-        match u32::from(raw) {
-            zmq_sys::ZMQ_EVENT_CONNECTED => CONNECTED,
-            zmq_sys::ZMQ_EVENT_CONNECT_DELAYED => CONNECT_DELAYED,
-            zmq_sys::ZMQ_EVENT_CONNECT_RETRIED => CONNECT_RETRIED,
-            zmq_sys::ZMQ_EVENT_LISTENING => LISTENING,
-            zmq_sys::ZMQ_EVENT_BIND_FAILED => BIND_FAILED,
-            zmq_sys::ZMQ_EVENT_ACCEPTED => ACCEPTED,
-            zmq_sys::ZMQ_EVENT_ACCEPT_FAILED => ACCEPT_FAILED,
-            zmq_sys::ZMQ_EVENT_CLOSED => CLOSED,
-            zmq_sys::ZMQ_EVENT_CLOSE_FAILED => CLOSE_FAILED,
-            zmq_sys::ZMQ_EVENT_DISCONNECTED => DISCONNECTED,
-            zmq_sys::ZMQ_EVENT_MONITOR_STOPPED => MONITOR_STOPPED,
-            zmq_sys::ZMQ_EVENT_HANDSHAKE_FAILED_NO_DETAIL => HANDSHAKE_FAILED_NO_DETAIL,
-            zmq_sys::ZMQ_EVENT_HANDSHAKE_SUCCEEDED => HANDSHAKE_SUCCEEDED,
-            zmq_sys::ZMQ_EVENT_HANDSHAKE_FAILED_PROTOCOL => HANDSHAKE_FAILED_PROTOCOL,
-            zmq_sys::ZMQ_EVENT_HANDSHAKE_FAILED_AUTH => HANDSHAKE_FAILED_AUTH,
-            zmq_sys::ZMQ_EVENT_ALL => ALL,
-            x => panic!("unknown event type {}", x),
-        }
+bitflags! {
+    /// Socket Events
+    /// A bitmask of the socket events you wish to monitor
```

Comment:
Note that documentation lines without empty line in between are treated as a single sentence/paragraph: this'll read weird as `Socket Events A bitmask of the socket events you wish to monitor` :wink: